### PR TITLE
feat(reporter): add opt-in EARL reporter

### PR DESCRIPTION
**Pull request by Garbee**
This pull request adds a new "earl" reporter to axe-core, enabling output of accessibility test results in the W3C EARL (Evaluation and Report Language) JSON-LD format. This makes it easier to integrate axe-core results with tools and workflows that consume standardized accessibility reports. The implementation includes mapping of rule tags to WCAG Success Criteria and comprehensive tests for the new reporter.

**New EARL Reporter Integration:**

* Added a new "earl" reporter that outputs results as a W3C EARL JSON-LD document, including a canonical `@context` and a `@graph` of assertions. The reporter is opt-in and can be set globally or per run. 
**EARL Assertion and WCAG Mapping Utilities:**

* Implemented `earlAssertion` to create EARL Assertion objects, including mapping rule tags to WCAG 2.1 and earlier Success Criteria for the `isPartOf` field, and handling various result outcomes. 
* Defined the EARL JSON-LD context and WCAG mapping data for use in the reporter.

**Testing:**

* Added comprehensive tests for the new EARL reporter, covering registration, output structure, assertion generation, WCAG mapping, version handling, and edge cases.

### CODE REVIEW

File: lib/core/reporters/earl/earl-assertion.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+ * @throws {TypeError} when `source` is not a string
+ */
+export default function earlAssertion({
+  source,
+  version,
+  ruleId,
+  tags = [],
+  mode = 'automatic',
+  outcome = 'untested'
+}) {
```
[the remark this answers is not in the document]
Comment (Garbee) [reply lib/core/reporters/earl/earl-assertion.js:24]: Addressed in 643d2475: `version` is now documented as required (the `[optional]` annotation was misleading — it is used unconditionally to build `assertedBy` and the Deque University `@id`). The reporter always passes `axe.version`, and `earlAssertion` is a private helper, so coupling it to the `axe` global was avoided in favor of making the real contract explicit.

File: lib/core/reporters/earl/index.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+  // Mirror the raw reporter: a non-array first argument (e.g. an error object
+  // the results pipeline forwarded) is passed through untouched, not crashed on.
+  if (!results || !Array.isArray(results)) {
+    return callback(results);
+  }
+
+  const { environmentData } = options;
+  const source = getEnvironmentData(environmentData).url ?? '';
+  const version = axe.version;
```
[the remark this answers is not in the document]
Comment (Garbee) [reply lib/core/reporters/earl/index.js:39]: Addressed in 643d2475: `url` is now normalized with a `typeof` check (`typeof reportedUrl === 'string' ? reportedUrl : ''`) before it becomes the assertion source, so a non-string `url` from a passthrough `environmentData` yields an empty-string source instead of throwing in `earlAssertion`. Added a test covering both the absent-url and non-string-url cases.

**Comment by padmavemulapati** [comment]
Verified in https://github.com/dequelabs/axe-devtools-reporter/issues/468
